docs/ref/forms/renderers.txt

```diff
@@ -0,0 +1,161 @@
+======================
+The form rendering API
+======================
+
+.. versionadded:: 1.11
+
+Django form widgets are rendered using the Django template backend system.
+
+The form rendering process can be customized at several levels:
+
+* Widgets can specify custom template names.
+* Forms and widgets can specify custom renderer classes.
+* Built-in templates can be overridden within a project or application.
+
+.. _low-level-widget-render-api:
+
+The low-level render API
+========================
+
+Widget templates are stored in the ``django/forms/widgets`` path. A project
+can provide a custom template for ``input.html`` by defining
+``django/forms/widgets/input.html``, for example.
+
+The rendering of form templates is controlled by a customizable renderer class.
+A custom renderer can be specified by updating the :setting:`FORM_RENDERER`
+setting. It defaults to
+``'``:class:`django.forms.renderers.templates.DjangoTemplateRenderer`\ ``'``.
+
+.. module:: django.forms.renderers.templates
+
+.. class:: BaseTemplateRenderer
```

Comment:
I'm not sure that we should document `BaseTemplateRenderer` or `EngineRendererMixin`; they seem like internal implementation details to me. I think any hypothetical value they'd provide to the implementor of a custom renderer (which is minor, they don't do very much) is outweighed by the added "ravioli code" complexity introduced by inheriting from built-in mixins and base classes. If the built-in renderers don't suit, the implementor of a custom renderer is likely to need to modify the code provided by these classes anyway, and I think we do them a better service by encouraging them to just implement what they need directly.

Documenting these also reduces our refactoring flexibility in the future.